### PR TITLE
Shellcheck & yamllint in Travis

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Add shellcheck travis. Still fails because still many warning/info/style issues. Not fixing them right now.

### CODE REVIEW

**Comment by ar13pit** [comment]
This is an interesting use of Travis :)

File: ack/setup [hunk 1]
```diff
@@ -1,3 +1,3 @@
-#!/bin/bash
+#! /usr/bin/env bash
```
Comment (jlunenburg) [inline ack/setup]: Why? Did you check, e.g., https://unix.stackexchange.com/questions/29608/why-is-it-better-to-use-usr-bin-env-name-instead-of-path-to-name-as-my ?

Comment (MatthijsBurgh) [reply ack/setup]: To have it to be consistent. As in your link, one is not better than the other. It is or a static path (`/bin/bash`) or one which is resolved based on your path (`/usr/bin/env bash`)

Comment (ar13pit) [reply ack/setup]: IMHO `/usr/bin/env` based formulations are more generalized. The effect can be seen predominantly in a virtualenv setting for python2 and python3. Having just `/usr/bin/env python` automatically resolves the executable based on the environment. 

However I have no knowledge of whether something like virtualenv can be set for something like bash but I like this formulation better.

Comment (MatthijsBurgh) [reply ack/setup]: @ar13pit if you approve? Could review the PR?

**Comment by ar13pit** [comment]
@MatthijsBurgh I think you should merge this one and open a new PR for the remaining targets

**Comment by MatthijsBurgh** [comment]
All remaining issues are reported in #48

**Pull request merged**